File: src/#199/solution.js
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val
  this.left = left === undefined ? null : left
  this.right = right === undefined ? null : right
}

var rightSideView = function (root) {
  let arr = []

  function traversal(root, level) {
    if (!root) {
      return
    }
    if (arr.length === level) {
      arr.push(root.val)
    }
    traversal(root.right, level + 1)
    traversal(root.left, level + 1)
  }

  traversal(root,0)


  return arr
}

File: src/#199/solution.go
package solution

type TreeNode struct {
	Val         int
	Left, Right *TreeNode
}

func rightSideView(root *TreeNode) []int {
	ans := []int{}
	traversal(root, &ans, 0)
	return ans
}

func traversal(root *TreeNode, arr *[]int, level int) {
	if root == nil {
		return
	}
	if level == len(*arr) {
		*arr = append(*arr, root.Val)
	}
	traversal(root.Right, arr, level+1)
	traversal(root.Left, arr, level+1)
}
